### PR TITLE
Viewer suggested weapon additions

### PR DESCRIPTION
**Pull request by Dweblenod**
Three new chainsaw suggested and designed by The Elegant One and one suggested by BigHomoBug designed by Dweblenod, each with lipstick alternatives

-Thistleblower: faster than the rest, made with chainsaw and rose bush
-Emerald Immolator: fire damage, made with chainsaw and emerald sword
-Frosttooth: ice shard effect, made with emerald immolator and club zero
-Obsidiator: made with frosttooth and obsidian

Added knives subsection of swords, with new OnHitEffect property called backstab. All suggested and designed by Addie

-Dagger: made by two iron ingots and a stick
-Nife: has Sord drop effect and no backstab(sbahj), made with dagger and sbahj poster
-Light of my Knife: made with dagger and lantern
-Starshard Tri-Blade: more than normal backstab damage
-Toothripper: made with dagger and rotten flesh/bone

An electric autoharp is added which can be bought from consorts and plays one of 9 random major chords recorded by Dweblenod

### CODE REVIEW

**Comment by kirderf1** [comment]
There are conflicts that need to be resolved. Can you resolve them on your branch?

**Comment by Dweblenod** [comment]
> There are conflicts that need to be resolved. Can you resolve them on your branch?

Working on it now! Also including new weapons made by Addie

**Comment by kirderf1** [comment]
As a tip, you've got a backlog of commits on your 1.15 that on mraof/1.15 was squashed into single commits
You could try and reset your 1.15 branch to the commit on mraof/1.15 to get rid of them
After that you could rebase it onto mraof/1.15 whenever you want to keep it up with commits on mraof, and avoid commiting directly to your branch to avoid creating a new backlog

File: src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java [hunk 1]
```diff
@@ -238,6 +239,35 @@ static OnHitEffect armorBypassingDamageMod(float additionalDamage, EnumAspect as
...
+					target.world.addParticle(ParticleTypes.DAMAGE_INDICATOR, true, target.getPosition().add(reversedBackVec).getX(), target.getEyePosition(1F).y - 1, target.getPosition().add(reversedBackVec).getZ(), 0.1,0.1,0.1);
+				}
+				
+				DamageSource source;
+				if(attacker instanceof PlayerEntity)
+					source = DamageSource.causePlayerDamage((PlayerEntity) attacker);
+				else source = DamageSource.causeMobDamage(attacker);
+				
+				Item usedItem = stack.getItem();
+				if(usedItem instanceof WeaponItem)
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java:263]: This check seems unneccesary

Comment (Dweblenod) [reply src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java]: Should be fixed now!

**Pull request merged**